scene/main/node.cpp

```diff
@@ -1715,8 +1715,50 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = false;
+	Node *search = p_parent;
+	Node *owner_temp = data.owner;
+	List<Node *> common_parents;
+
+	//First determine if the original owner is an ancestor of the node we're reparenting to
+	while (search) {
+		if (search == data.owner) {
+			preserve_owner = true;
+			break;
+		}
+		search = search->data.parent;
+	}
+
+	//If so, find all children of the reparented node who have the same owner so that we can reassign them
```

Comment:
```suggestion
	// If so, find all children of the reparented node who have the same owner so that we can reassign them.
```
Same with other comments.